go/worker/common/p2p/peermgmt.go

```diff
@@ -113,21 +155,36 @@ func (mgr *PeerManager) updateNodeLocked(node *node.Node, peerID core.PeerID) {
 	// way, then don't bother doing anything.
 	if oldNode := mgr.peers[peerID]; oldNode != nil {
 		if oldNode.addrHash.Equal(&addrHash) {
-			return
+			// But, if the peer isn't connected for some reason, still try to reconnect to it.
+			if mgr.host.Network().Connectedness(peerID) == network.Connected {
+				mgr.logger.Debug("updateNodeLocked addresses unchanged and peer still connected",
```

Comment:
Since the log entry will include the line number this can be just:
```suggestion
				mgr.logger.Debug("addresses unchanged and peer still connected",
```